### PR TITLE
Avoid robolectric added add-opens on JDK 1.8

### PR DESCRIPTION
**Pull request by yschimke**
Add add-opens and also fails to run.

### CODE REVIEW

**Comment by yschimke** [comment]
See https://github.com/square/okhttp/actions/runs/12560902710/job/35018955029

```
> Task :okhttp:testDebugUnitTest FAILED
Unrecognized option: --add-opens=java.base/java.io=ALL-UNNAMED
Error: Could not create the Java Virtual Machine.
```

**Comment by yschimke** [comment]
Still 1/3 of JDK usage https://newrelic.com/sites/default/files/2023-04/new-relic-2023-state-of-the-java-ecosystem-2023-04-20.pdf

And

> The end date for Oracle's Java 8 LTS (Long Term Support) is December 2030: 
Premier support: Ended on March 31, 2022
Extended support: Available until December 2030 with an additional fee

**Comment by yschimke** [comment]
Shouldn't be firing!

https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/tasks/factory/AndroidUnitTest.java;l=339?q=robolectric%20add-opens&start=11

```
            if (task.getJavaLauncher()
                            .get()
                            .getMetadata()
                            .getLanguageVersion()
                            .compareTo(JavaLanguageVersion.of(17))
                    >= 0) {
                task.jvmArgs((ImmutableList.of("--add-opens=java.base/java.io=ALL-UNNAMED")));
            }
```

File: okhttp/build.gradle.kts [hunk 1]
```diff
@@ -273,6 +273,19 @@ configure<CheckstyleExtension> {
   sourceSets = listOf(project.sourceSets["main"])
 }
 
+afterEvaluate {
+  tasks.withType<Test> {
+    if (javaLauncher.get().metadata.languageVersion.asInt() < 9) {
+      // Work around robolectric requirements and limitations
```
Comment (swankjesse) [inline okhttp/build.gradle.kts:279]: Nice comment!

**Review by swankjesse — COMMENTED** [review]
LGTM

**Pull request merged**